source/common/http/filter/ratelimit.cc

```diff
@@ -11,8 +11,9 @@
 namespace Http {
 namespace RateLimit {
 
-const Http::HeaderMapPtr Filter::TOO_MANY_REQUESTS_HEADER{new Http::HeaderMapImpl{
-    {Http::Headers::get().Status, std::to_string(enumToInt(Code::TooManyRequests))}}};
+const std::unique_ptr<const Http::HeaderMap> Filter::TOO_MANY_REQUESTS_HEADER{
+    new Http::HeaderMapImpl{
```

Comment:
Yes, for init, we need to more careful, and we do handle more complex cases in init by having functions that return status variables. I will do a doc update.